Apps/System/menu.c

```diff
@@ -1,222 +1,139 @@
 /* menu.c
- * Poor mans menu to stop gap us
- * 
+ *
+ * Generic menu component.
+ *
  * RebbleOS
  * 
  * Author: Barry Carter <barry.carter@gmail.com>.
  */
 
 #include "rebbleos.h"
-#include "librebble.h"
 #include "menu.h"
-#include "appmanager.h"
-#include "ngfxwrap.h"
-
-extern void graphics_draw_bitmap_in_rect(GContext*, GBitmap*, GRect);
-extern void flash_dump(void);
-
-/*
-// menu
-
-This is teh suckiest. All hard coded (in the worst way) until we get proper menu support
-coded up
-
-*/
-
-#define STANDARD_MENU_COUNT 4
 
-static int8_t _menu_index = 0;
-MenuItem _main_menu[4];
-static char *_selected_menu_name;
-static uint8_t _app_count = 0;
-
-#define MENU_MAIN       0
-#define MENU_WATCH      1
-#define MENU_CONSOLE    2
+MenuItems* menu_items_create(uint16_t capacity)
+{
+    MenuItems *result = (MenuItems *) calloc(1, sizeof(MenuItems));
```

Comment:
yeah I think this does work right now, but I think that eventually we're going to want to move this to the app heap; can you mark all the allocation sites here with 'XXX fix when we have an app heap'